lib/file.js

```diff
@@ -103,6 +105,141 @@ async function downloadFile(url, filepath, options) {
     }, options);
 }
 
+/**
+ * Download file using multiple simultaneous transfers
+ * Throws the first unrecoverable error if unsuccessful, all others are logged
+ * @param {String|URL} url Location of file to download (e.g. presigned URL) 
+ * @param {String} filepath Path to save file
+ * @param {DownloadFileOptions} options Additional options such as headers, retry features, etc 
+ * @returns {Promise} resolves when download completes
+ */
+async function downloadFileConcurrently(url, filepath, options) {
+    const downloader = new BlockDownload();
+    return downloader.downloadFiles({
+        downloadFiles: [
+            {
+                fileUrl : url,
+                filePath : filepath,
+                fileSize : -1        
+            }
+        ],
+        ...options
+    });
+}
+
+/**
+ * @typedef {Object} UploadFileOptions
+ *
+ * @property {String} [method] Optional HTTP method (defaults to 'PUT')
+ * @property {Number} [timeout] Optional socket timeout
+ * @property {Object} [headers] Optional override of request headers
+ * @property {Number} [retryMaxDuration=60000] time to retry until throwing an error (ms)
+ * @property {Number} [retryMaxCount=5] time to retry until throwing an error (ms)
+ * @property {Number} [retryInterval=100] time between retries, used by exponential backoff (ms)
+ * @property {Boolean} [retryEnabled=true] retry on failure enabled
+ * @property {Boolean} [retryAllErrors=false] whether or not to retry on all http error codes or just >=500
+ * @property {Number} [maxConcurrent] Optional concurrent amount.
+ */
+/**
+ * Upload file using multiple simultaneous transfers
+ * Throws the first unrecoverable error if unsuccessful, all others are logged
+ * @param {String|URL} url Location to upload (e.g. presigned URL) 
+ * @param {String} filepath Path to locally saved file
+ * @param {UploadFileOptions} options Additional options such as headers, retry features, etc 
+ * @returns {Promise} resolves when upload completes
+ */
+async function uploadFileConcurrently(filepath, url, options) {
+    // options contains header information
+    const { size } = await getFileStats(filepath);
+
+    // extract upload options
+    const uploadOptions = filterObject(
+        options || {},
+        [ 'method', 'timeout', 'headers', 'maxConcurrent', 'concurrent',
+            'retryMaxDuration', 'retryInterval', 'retryEnabled',
+            'retryAllErrors', 'retryMaxCount', 'preferredPartSize' ]
+    );
```

Comment:
we should match up the retryOptions with the ones that are supported: https://github.com/adobe/node-httptransfer/blob/6e588e161cb47541d5b5d13aa5a2db6522631e02/lib/retry.js#L79-L85
(with the expection of `timeout` for `socketTImeout` since both are supported: https://github.com/adobe/node-httptransfer/blob/6e588e161cb47541d5b5d13aa5a2db6522631e02/lib/retry.js#L52